Src/Support/Google.Apis/Requests/BatchRequest.cs

```diff
@@ -179,7 +179,35 @@ public async Task ExecuteAsync(CancellationToken cancellationToken)
             var result = await httpClient.PostAsync(new Uri(batchUrl), outerContent, cancellationToken)
                 .ConfigureAwait(false);
 
-            result.EnsureSuccessStatusCode();
+            if (!result.IsSuccessStatusCode)
+            {
+                try
+                {
+                    // Parse the error from the current response.
+                    var error = await service.DeserializeError(result).ConfigureAwait(false);
+                    if (error != null)
+                    {
+                        throw new GoogleApiException(service.Name, error.ToString())
+                        {
+                            Error = error,
+                            HttpStatusCode = result.StatusCode
+                        };
+                    }
+                }
+                catch(Exception apiException)
+                {
+                    try
+                    {
+                        result.EnsureSuccessStatusCode();
+                    }
+                    catch (HttpRequestException requestException)
```

Comment:
Is there nothing in the HttpRequestException other than the message which would be useful to include?

Reply:
Not really: https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/net/System/Net/Http/HttpResponseMessage.cs#L142